lib/OpenQA/Worker/Common.pm

```diff
@@ -335,7 +351,16 @@ sub register_worker {
     $worker_caps = _get_capabilities;
     $worker_caps->{host} = $hostname;
     $worker_caps->{instance} = $instance;
-    $worker_caps->{worker_class} = $worker_settings->{WORKER_CLASS} || "qemu_" .  $worker_caps->{cpu_arch};
+    if ($worker_settings->{WORKER_CLASS}) {
+        $worker_caps->{worker_class} =$worker_settings->{WORKER_CLASS};
```

Comment:
Now this is what I'm talking about. Technically absolutely correct, but the missing space is red herring for my human visual pattern recognition.